src/components/farm-details/UnifiedDataLogsModal.tsx

```
@@ -69,8 +134,17 @@ export function UnifiedDataLogsModal({
 }: UnifiedDataLogsModalProps) {
+  const { handleError, handleSuccess } = useErrorHandler()
   const [selectedDate, setSelectedDate] = useState(new Date().toISOString().split('T')[0])
   const [currentLogType, setCurrentLogType] = useState<LogType | null>(null)
+
+  // Memoize validation results for performance
+  const chemicalValidationResults = useMemo(() => {
+    return chemicalEntries.map(entry => ({
+      id: entry.id,
+      validation: validateChemicalEntry(entry)
+    }))
+  }, [chemicalEntries, validateChemicalEntry])
```

Biome (2.1.2)

[error] This variable is used before its declaration.

The variable is declared here:


(lint/correctness/noInvalidUseBeforeDeclaration)